Automattic/jetpack
# Send a Message / WhatsApp block

**Note: New commits require the new files added in this PR to be re-added to the Fusion generated diff (D44901-code) as the WordPress.com gating (D44978-code) for the block depends on the resulting build**

This block provides two things. The first is a parent "Send a Message" block that is the container for all messaging service inner blocks (very similar to the social icons block). The second is a WhatsApp Button block that provides the primary functionality for connecting to WhatsApp.

So far the "Send a Message" block is limited only to WhatsApp as a service, so this inner block is inserted by default. Going forward we can easily add more services.

There is a default "WhatsApp Button" variation of the "Send a Message" block that means that "WhatsApp Button" shows up in the inserter, not "Send a Message". In the future we can also provide other service variations so they appear in the inserter, rather than a generic "Send a Message" block.

This is how it works (note on mobile phones, tapping the button will open WhatsApp directly):

![2020-05-26 09 39 57](https://user-images.githubusercontent.com/1464705/82927158-1e615600-9f35-11ea-9358-15c62d61dead.gif)

In this version of the block, I have hidden the child block inserter to the right of the WhatsApp button that would be used to insert buttons for other services. There is only one service, so it's not useful.

One caveat -- variations do not work via the `/` quick inserter, so you will not see the "WhatsApp Button" block appear here. This is being tracked here and I'll see what I can do to move this along: https://github.com/WordPress/gutenberg/issues/20583

#### Testing instructions:
* Insert a WhatsApp button
* Add a phone number that is associated with a WhatsApp account
* Publish the post
* Confirm that clicking the button opens WhatsApp and the phone number and initial message look correct.
* Try on a mobile device and confirm that WhatsApp opens correctly.
* Adjust the color settings and confirm that this changes correctly.
* Insert multiple buttons on the page and confirm they all work separately and correctly.

#### Proposed changelog entry for your changes:

* Introduced a new WhatsApp button block to allow your users to send you messages on WhatsApp from your website.
